// @theme-ui/preset-polaris/dist/index.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@theme-ui/preset-base')) :
  typeof define === 'function' && define.amd ? define(['exports', '@theme-ui/preset-base'], factory) :
  (factory((global.presetPolaris = {}),global.base));
}(this, (function (exports,base) {
  base = base && base.hasOwnProperty('default') ? base['default'] : base;

  function _defineProperty(obj, key, value) {
    if (key in obj) {
      Object.defineProperty(obj, key, {
        value: value,
        enumerable: true,
        configurable: true,
        writable: true
      });
    } else {
      obj[key] = value;
    }

    return obj;
  }

  function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);

    if (Object.getOwnPropertySymbols) {
      var symbols = Object.getOwnPropertySymbols(object);
      if (enumerableOnly) symbols = symbols.filter(function (sym) {
        return Object.getOwnPropertyDescriptor(object, sym).enumerable;
      });
      keys.push.apply(keys, symbols);
    }

    return keys;
  }

  function _objectSpread2(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i] != null ? arguments[i] : {};

      if (i % 2) {
        ownKeys(Object(source), true).forEach(function (key) {
          _defineProperty(target, key, source[key]);
        });
      } else if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
      } else {
        ownKeys(Object(source)).forEach(function (key) {
          Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
      }
    }

    return target;
  }

  var text = 'system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", sans-serif';
  var polaris = _objectSpread2({}, base, {
    colors: {
      text: '#454f5b',
      background: '#fff',
      primary: '#5c6ac4',
      secondary: '#006fbb',
      highlight: '#47c1bf',
      muted: '#e6e6e6',
      gray: '#dfe3e8',
      accent: '#f49342',
      darken: '#00044c',
      modes: {
        dark: {
          text: '#3e4155',
          background: '#000639',
          primary: '#9c6ade',
          secondary: '#b4e1fa',
          highlight: '#b7ecec',
          muted: '#e6e6e6'
        }
      }
    },
    fonts: {
      body: text,
      heading: text,
      moonospace: 'Menlo, monospace'
    },
    fontWeights: {
      body: 400,
      heading: 600,
      bold: 700
    },
    lineHeights: {
      body: 1.75,
      heading: 1.25
    }
  });

  exports.polaris = polaris;
  exports.default = polaris;

})));
