// @theme-ui/preset-funk/dist/index.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@theme-ui/preset-base')) :
  typeof define === 'function' && define.amd ? define(['exports', '@theme-ui/preset-base'], factory) :
  (factory((global.presetFunk = {}),global.base));
}(this, (function (exports,base) {
  base = base && base.hasOwnProperty('default') ? base['default'] : base;

  function _defineProperty(obj, key, value) {
    if (key in obj) {
      Object.defineProperty(obj, key, {
        value: value,
        enumerable: true,
        configurable: true,
        writable: true
      });
    } else {
      obj[key] = value;
    }

    return obj;
  }

  function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);

    if (Object.getOwnPropertySymbols) {
      var symbols = Object.getOwnPropertySymbols(object);
      if (enumerableOnly) symbols = symbols.filter(function (sym) {
        return Object.getOwnPropertyDescriptor(object, sym).enumerable;
      });
      keys.push.apply(keys, symbols);
    }

    return keys;
  }

  function _objectSpread2(target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i] != null ? arguments[i] : {};

      if (i % 2) {
        ownKeys(Object(source), true).forEach(function (key) {
          _defineProperty(target, key, source[key]);
        });
      } else if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
      } else {
        ownKeys(Object(source)).forEach(function (key) {
          Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
      }
    }

    return target;
  }

  var funk = _objectSpread2({}, base, {
    fonts: {
      body: 'Poppins, sans-serif',
      heading: 'Poppins, sans-serif',
      monospace: 'Menlo, monospace'
    },
    lineHeights: {
      body: 1.625,
      heading: 1.25
    },
    fontWeights: {
      body: 400,
      heading: 900,
      bold: 700
    },
    colors: _objectSpread2({}, base.colors, {
      primary: '#609',
      secondary: '#306'
    }),
    styles: _objectSpread2({}, base.styles)
  });

  exports.funk = funk;
  exports.default = funk;

})));
