// @theme-ui/presets/dist/index.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@theme-ui/preset-base'), require('@theme-ui/preset-dark'), require('@theme-ui/preset-deep'), require('@theme-ui/preset-funk'), require('@theme-ui/preset-future'), require('@theme-ui/preset-roboto'), require('@theme-ui/preset-swiss'), require('@theme-ui/preset-system'), require('@theme-ui/preset-tosh'), require('@theme-ui/preset-bootstrap'), require('@theme-ui/preset-bulma'), require('@theme-ui/preset-polaris'), require('@theme-ui/preset-tailwind')) :
	typeof define === 'function' && define.amd ? define(['exports', '@theme-ui/preset-base', '@theme-ui/preset-dark', '@theme-ui/preset-deep', '@theme-ui/preset-funk', '@theme-ui/preset-future', '@theme-ui/preset-roboto', '@theme-ui/preset-swiss', '@theme-ui/preset-system', '@theme-ui/preset-tosh', '@theme-ui/preset-bootstrap', '@theme-ui/preset-bulma', '@theme-ui/preset-polaris', '@theme-ui/preset-tailwind'], factory) :
	(factory((global.presets = {}),global.presetBase,global.presetDark,global.presetDeep,global.presetFunk,global.presetFuture,global.presetRoboto,global.presetSwiss,global.presetSystem,global.presetTosh,global.presetBootstrap,global.presetBulma,global.presetPolaris,global.presetTailwind));
}(this, (function (exports,presetBase,presetDark,presetDeep,presetFunk,presetFuture,presetRoboto,presetSwiss,presetSystem,presetTosh,presetBootstrap,presetBulma,presetPolaris,presetTailwind) {
	presetBase = presetBase && presetBase.hasOwnProperty('default') ? presetBase['default'] : presetBase;
	presetDark = presetDark && presetDark.hasOwnProperty('default') ? presetDark['default'] : presetDark;
	presetDeep = presetDeep && presetDeep.hasOwnProperty('default') ? presetDeep['default'] : presetDeep;
	presetFunk = presetFunk && presetFunk.hasOwnProperty('default') ? presetFunk['default'] : presetFunk;
	presetFuture = presetFuture && presetFuture.hasOwnProperty('default') ? presetFuture['default'] : presetFuture;
	presetRoboto = presetRoboto && presetRoboto.hasOwnProperty('default') ? presetRoboto['default'] : presetRoboto;
	presetSwiss = presetSwiss && presetSwiss.hasOwnProperty('default') ? presetSwiss['default'] : presetSwiss;
	presetSystem = presetSystem && presetSystem.hasOwnProperty('default') ? presetSystem['default'] : presetSystem;
	presetTosh = presetTosh && presetTosh.hasOwnProperty('default') ? presetTosh['default'] : presetTosh;
	presetBootstrap = presetBootstrap && presetBootstrap.hasOwnProperty('default') ? presetBootstrap['default'] : presetBootstrap;
	presetBulma = presetBulma && presetBulma.hasOwnProperty('default') ? presetBulma['default'] : presetBulma;
	presetPolaris = presetPolaris && presetPolaris.hasOwnProperty('default') ? presetPolaris['default'] : presetPolaris;
	presetTailwind = presetTailwind && presetTailwind.hasOwnProperty('default') ? presetTailwind['default'] : presetTailwind;



	exports.base = presetBase;
	exports.dark = presetDark;
	exports.deep = presetDeep;
	exports.funk = presetFunk;
	exports.future = presetFuture;
	exports.roboto = presetRoboto;
	exports.swiss = presetSwiss;
	exports.system = presetSystem;
	exports.tosh = presetTosh;
	exports.bootstrap = presetBootstrap;
	exports.bulma = presetBulma;
	exports.polaris = presetPolaris;
	exports.tailwind = presetTailwind;

})));
